### PR TITLE
avoid cyclic dependency between awful and menubar

### PR DESCRIPTION
**Pull request by actionless**
related to #1400

### CODE REVIEW

File: lib/awful/hotkeys_popup/widget.lua [hunk 1]
```diff
@@ -290,8 +290,8 @@ function widget.new()
                     joined_labels = joined_labels .. rendered_hotkey .. (i~=#_keys and "\n" or "")
                     end
                 current_column.layout:add(wibox.widget.textbox(joined_labels))
-                local max_width = compute_textbox_width(wibox.widget.textbox(max_label_content), s) +
-                                                    widget_instance.group_margin
+                local max_width, _ = wibox.widget.textbox(max_label_content):get_preferred_size(s)
+                max_width = max_width + widget_instance.group_margin
                 if not current_column.max_width or max_width > current_column.max_width then
```
Comment (psychon) [inline lib/awful/hotkeys_popup/widget.lua]: This part of the change seems unnecessary? The file can just be left as-is. (Also, I think luacheck will now complain that `compute_textbox_width` is set, but unused)

**Comment by psychon** [comment]
Ok, I'm confused. So this says "avoid cyclic dependency between awful and menubar". However, the change does not actually remove any kind of dependency, does it?

File: lib/awful/hotkeys_popup/widget.lua (hunk 1 above)
Comment (actionless) [inline lib/awful/hotkeys_popup/widget.lua]: my  bad, forgot to remove the require line

**Comment by blueyed** [comment]
Thanks!

**Comment by actionless** [comment]
@psychon do you mean table.pack() or there is some different way?

**Comment by psychon** [comment]
You mean the single line?
Isn't
```lua
local max_width, _ = wibox.widget.textbox(max_label_content):get_preferred_size(s)
max_width = max_width + widget_instance.group_margin
```
equivalent to
```
local max_width = wibox.widget.textbox(max_label_content):get_preferred_size(s) + widget_instance.group_margin
```
The only difference should be that the variable with name `_` isn't set.

Put differently: Doesn't Lua ignore all following values from multiple return values in binary operations?

**Comment by actionless** [comment]
thanks, i didn't actually knew about that:
```lua
> function f() return 1,2,3 end ; a,b=f() ; print(a,b,c)
1       2       nil
```

i was expecting it to throw an error in that case

**Pull request merged**